### ISSUE TITLE
Feature Request: Option to fill Company Discretionary Data field in NACHA output file

### ISSUE THREAD

**Issue by trustedcomputer** [agritheory/check_run#74]
JPMorgan Chase Bank requires the Company Discretionary Data field in the Batch Header be set in the NACHA output file be set to the funding account number.  Currently, in apps/check_run/check_run/check_run/doctype/check_run/check_run.py, this is set to a null value in line 580:

`company_discretionary_data='',`

Would it be possible to either create a field for this or to have the option to use the Bank Account document's bank_account_no field to fill this in the NACHA output file?

**Comment by trustedcomputer** [comment]
Reference:
![Batch-Header](https://user-images.githubusercontent.com/75872475/223566337-b20569ed-80d0-4b62-8ef3-7bd3b838d8c9.png)

**Comment by agritheory** [comment]
@put3r-r00t3r Hi Ben, In this design, would a static value populated from Check Run Settings be sufficient or does it need to be supplied at the time the file is generated?

**Comment by trustedcomputer** [comment]
@agritheory According to the [published specifications](https://achdevguide.nacha.org/ach-file-details), this data could be just about anything: "Originator/ODFI may include codes of significance only to them to enable specialized handling of all entries within the batch." My read on that is that the data is supposed to be batch-specific. So although I would not need it, there conceivably be use cases out there where people would need different values supplied for each Check Run document. Perhaps the best way would be a field in the Check Run document that is fetched from the Check Run Settings document, but still modifiable.

**Comment by agritheory** [comment]
@put3r-r00t3r 
> Perhaps the best way would be a field in the Check Run document that is fetched from the Check Run Settings document, but still modifiable.

To make this work best, I would add a field that is only displayed when `docstatus > 0` and has the 'allow_on_submit` attribute.

**Comment by trustedcomputer** [comment]
@agritheory I've got most of this done, but I'm not sure how to hide the new field until `docstatus > 0`. 

But I have tried to figure it out and here's what I know so far, examining existing code:
This will probably be done in the check_run.js file. 
It will probably look something like this:

```
refresh: frm => {
......
if (frm.doc.docstatus < 1 && frm.doc.__onload && frm.doc.__onload.errors) {
......
company_discretionary_data.hide()
....
```

Am I on the right track?

**Comment by agritheory** [comment]
@put3r-r00t3r I would do it with an `eval:doc.docstatus==1` in the 'depends on' field in the doctype. I think that's easier that doing it in the `refresh` function

**Comment by trustedcomputer** [comment]
Pull Request: https://github.com/agritheory/check_run/pull/77

**Comment by agritheory** [comment]
Resolved with #77
